dashboard/test/models/user_test.rb

```diff
@@ -4140,46 +4140,67 @@ def hide_scripts_in_sections(section1, section2)
     assert_equal teacher.user_school_infos.count, 2
   end
 
-  test 'can grant admin role with google oauth, codeorg account' do
+  test 'can grant admin role with only google oauth, codeorg account' do
     email = 'fernhunt@code.org'
-    migrated_teacher = create(:teacher, :with_google_authentication_option, email: email)
+    migrated_teacher = create(:teacher, :google_sso_provider, email: email)
 
+    assert_equal 1, migrated_teacher.authentication_options.count
     migrated_teacher.update!(admin: true)
 
     assert migrated_teacher.valid?
     assert migrated_teacher.errors[:admin].empty?
   end
 
-  test 'cannot grant admin role when google authentication option not does exist' do
-    email = 'annieeasley@code.org'
-    migrated_teacher = create(:teacher, email: email)
+  test 'cannot grant admin role when unmigrated teacher account' do
+    unmigrated_teacher_without_password = create :teacher, :demigrated
+    unmigrated_teacher_without_password.update_attribute(:encrypted_password, '')
+
+    assert_raises(ActiveRecord::RecordInvalid) do
+      unmigrated_teacher_without_password.update!(admin: true)
+    end
+
+    refute unmigrated_teacher_without_password.reload.admin?
```

Comment:
we should add the same error assertions to these tests as in the previous PR - [link](https://github.com/code-dot-org/code-dot-org/pull/31847#discussion_r346944255)